packages/bootstrap-4/src/TextWidget/TextWidget.tsx

```diff
@@ -36,23 +36,33 @@ const TextWidget = ({
 
   return (
     <Form.Group controlId={id} className="mb-0">
-      <Form.Label>
+      <Form.Label className={rawErrors.length > 0 ? "text-danger" : ""}>
         {label || schema.title}
-        {required ? "*" : null}
+        {(label || schema.title) && required ? "*" : null}
       </Form.Label>
       <Form.Control
         id={id}
         autoFocus={autofocus}
         required={required}
         disabled={disabled || readonly}
-        name={name}
+        className={rawErrors.length > 0 ? "is-invalid" : ""}
+        list={schema.examples ? `examples_${id}` : undefined}
         type={type || (schema.type as string)}
         value={value || value === 0 ? value : ""}
         onChange={_onChange}
         onBlur={_onBlur}
         onFocus={_onFocus}
         {...textFieldProps}
       />
+      {schema.examples ? (
+        <datalist id={`examples_${id}`}>
+          {(schema.examples as any)
```

Comment:
Can we avoid `as any` by improving the type definition? If not, this is fine, but not ideal.